web/src/api/images/upload_image.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  errorHasuraQuery,
+  errorNotAllowed,
+  errorResponse,
+} from "src/backend/errors";
+import { NextApiRequest, NextApiResponse } from "next";
+import * as yup from "yup";
+import { validateRequestSchema } from "src/backend/utils";
+import { getAPIServiceGraphqlClient } from "src/backend/graphql";
+import { getSdk as checkUserInAppDocumentSDK } from "@/api/images/graphql/checkUserInApp.generated";
+import { Session, getSession, withApiAuthRequired } from "@auth0/nextjs-auth0";
+import { S3Client } from "@aws-sdk/client-s3";
+import { createPresignedPost } from "@aws-sdk/s3-presigned-post";
+import { logger } from "@/lib/logger";
+
+export type ImageUploadResponse = {
+  url?: string;
+  success?: boolean;
+};
+
+const schema = yup.object({
+  app_id: yup.string().strict().required(),
+  image_type: yup
+    .string()
+    .strict()
+    .oneOf([
+      "logo_img",
+      "hero_image",
+      "showcase_img_1",
+      "showcase_img_2",
+      "showcase_img_3",
+    ])
+    .required(),
+});
+
+export type ImageUploadBody = yup.InferType<typeof schema>;
+// This endpoint takes an AppID and checks the user is on the team. If so it returns a signed URL to upload an image to S3.
+export const handleImageUpload = withApiAuthRequired(
+  async (req: NextApiRequest, res: NextApiResponse<ImageUploadResponse>) => {
+    try {
```

Comment:
FYI this would normally require CORS but given you're changing it to Hasura custom actions we should be fine